R/utils.R

```diff
@@ -127,6 +127,41 @@ parse_answer = function(api_answer, multiple = FALSE, key = NULL) {
   }
 }
 
+#' Parse publisher list
+#'
+#' @param api_answer
+#'
+#' @return a parsed data.frame of publisher
+parse_publisher = function(api_answer) {
+  if (http_error(api_answer)) {
+    stop("The API endpoint could not be reached. Please try again later.")
+  }
+
+  xml_source = content(api_answer, encoding = "ISO-8859-1")
+
+  apicontrol = xml_text(xml_find_all(xml_source, "//apicontrol"))
+
+  if (apicontrol == "invalidkey") {
+    stop("The provided API key is invalid. ",
+         "You can register for a free API at ",
+         "http://www.sherpa.ac.uk/romeo/apiregistry.php")
+  }
+
+  publisher   = xml_text(xml_find_all(xml_source, "//name"))
+  romeocolour = xml_text(xml_find_all(xml_source, "//romeocolour"))
+  preprint    = xml_text(xml_find_all(xml_source, "//prearchiving"))
+  postprint   = xml_text(xml_find_all(xml_source, "//postarchiving"))
+  pdf         = xml_text(xml_find_all(xml_source, "//pdfarchiving"))
+  pdf         = ifelse(pdf == "unknown", NA_character_, pdf)
+
+  data.frame(publisher   = publisher,
+             romeocolour = romeocolour,
+             preprint    = preprint,
+             postprint   = postprint,
+             pdf         = pdf)
```

Comment:
Why not add embargo policies? For an example, you can try querying `yellow`.

It would much more easier if `parse_answer()` and `parse_publisher()` have similar outputs

Reply:
That may be an issue for some of them though...

http://www.sherpa.ac.uk/romeo/api29.php?pub=Inderscience&qtype=exact

Reply:
Ok will do and return NA if empty

Reply:
Woops with the last version we need to take into account these special cases...